src/stores/user-store.js

```diff
@@ -52,21 +53,78 @@ class UserStore extends EventEmitter {
         }
     }
 
+    _findUser (userInfo) {
+        const hasMatchedFields = (item, name) => item[name].toLowerCase().includes(userInfo[name].trim().toLowerCase());
+
+        return this._users.filter(anyUser => {
+            return hasMatchedFields(anyUser, 'name') &&
+                   hasMatchedFields(anyUser, 'phone') &&
+                   hasMatchedFields(anyUser, 'email') &&
+                   hasMatchedFields(anyUser, 'website');
+        });
+    }
+
+    _defineSearchFields () {
+        return {
+            name:    !!this._searchedUser.name.trim(),
+            phone:   !!this._searchedUser.phone.trim(),
+            email:   !!this._searchedUser.email.trim(),
+            website: !!this._searchedUser.website.trim()
+        };
+    }
+
+    _checkSearchModeAndEmit (emit) {
+        if (this._searchedUser instanceof UserInfo) {
+            const _foundUsers = this._findUser(this._searchedUser);
+
+            this.emit(EVENT_TYPE.usersFound, _foundUsers, this._defineSearchFields());
+        }
+        else {
+            this.emit(emit);
+        }
+    }
+
+    _actionAddNewUser (user, force) {
+        const index = this._findUserIndexByName(user);
+
+        if (index === -1 || force) {
+            this._addNewUser(user);
+            this._checkSearchModeAndEmit(EVENT_TYPE.userAdded);
+        }
+        else {
+            const userId = this._users[index].id;
+            this.emit(EVENT_TYPE.addingFailed, userId);
+        }
+    }
+
+    _actionUpdateUser (user) {
+        this._updateUser(user);
+        this._checkSearchModeAndEmit(EVENT_TYPE.change);
+    }
+
+    _actionFindUser (userInfo) {
+        this._searchedUser = userInfo;
```

Comment:
it's better to rename `this._searchedUser` -> `_searchedInfo`
And it's ok to set it with `null` when we need to clean searching